src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -754,6 +828,28 @@ public void emitNodeMetrics(long currWindowStartTime, MetricsDB metricsDB) throw
     }
   }
 
+  @VisibleForTesting
+  public void readBatchMetricsEnabledFromConf() {
```

Comment:
VisibleForTesting "Annotates a program element that exists, or is more widely visible than otherwise necessary, only for use in test code". This method would be private if we only needed it in trimOldSnapshots. However, we need it for testing purposes, so we make it VisibleForTesting.

Reply:
One of the ways to avoid it, if we at all want to test private methods, would be to encapsulate all such methods, in a wrapper class of your own. Then make the methods inside the wrapper as public and hence testable. The wrapper can be used in the main class as private or its getters/setters as visible for testing.

Reply:
Will address this by making the target methods private and exposing them via VisibleForTesting shims.